props/multi_prop.h

```diff
@@ -0,0 +1,501 @@
+template<class Saber, class Blaster, class Detonator, class Jetpack/*, class MorseCode , class Droid, class Vehicle*/>
+class MultiProp : public virtual Saber, public virtual Blaster, public virtual Detonator, public virtual Jetpack/*, public virtual MorseCode, public virtual Droid, public virtual Vehicle*/ {
+// One prop to rule them all, and in your config bind them!
+public:
+  const char* name() override { return "MultiProp"; }
+
+  // Button mapping for 2 and 3-button setups
+    // The "Blaster mapping"
+  uint32_t map_button(uint32_t b) {
+    switch (b) {
+#if NUM_BUTTONS == 3
+      case BUTTON_AUX:   return BUTTON_FIRE;
+      case BUTTON_AUX2:  return BUTTON_MODE_SELECT;
+#else
+      case BUTTON_POWER: return BUTTON_FIRE;
+      case BUTTON_AUX:   return BUTTON_MODE_SELECT;
+#endif
+      default: return b;
+    }
+  }
+
+    // The "return to normal mapping"
+  uint32_t reverse_map_button(uint32_t b) {
+    switch (b) {
+#if NUM_BUTTONS == 3
+      case BUTTON_FIRE:        return BUTTON_AUX;
+      case BUTTON_MODE_SELECT: return BUTTON_AUX2;
+#else
+      case BUTTON_FIRE:        return BUTTON_POWER;
+      case BUTTON_MODE_SELECT: return BUTTON_AUX;
+#endif
+      default: return b;
+    }
+  }
+
+  // Event handling for button presses, including combos for switching modes
+  bool Event(enum BUTTON button, EVENT event) override {
+    static bool powerPressed_ = false;    // Tracks BUTTON_POWER press state
+    static bool auxPressed_ = false;      // Tracks BUTTON_AUX   press state
+    if (event == EVENT_PRESSED) {
+      if (button == BUTTON_POWER) {
+        powerPressed_ = true;             // BUTTON_POWER is pressed
+      } else {
+        if (button == BUTTON_AUX) {
+        auxPressed_ = true;               // BUTTON_AUX   is pressed
+        }
+      }
+      // Check if both buttons are pressed simultaneously or near-simultaneously
+      if (powerPressed_ && auxPressed_ && holdStartTime == 0) {
+        holdStartTime = millis();         // Start the hold timer when both buttons are pressed
+        PVLOG_NORMAL << "Dual Long Push Initiated.\n";
+      }
+    } else {
+      if (event == EVENT_RELEASED) {
+        if (button == BUTTON_POWER) {
+          powerPressed_ = false;          // BUTTON_POWER is released
+        } else {
+          if (button == BUTTON_AUX) {
+            auxPressed_ = false;          // BUTTON_AUX   is released
+          }
+        }
+      }
+      //  Reset  the  hold  state  if  either  button  is released
+      if (!powerPressed_ || !auxPressed_) {
+        holdStartTime = 0;
+        PVLOG_NORMAL << "One of the two (or both) button(s) was (were) released too early!\n";
+      }
+    }
+    // Check if both buttons have been held for the required duration
+    if (powerPressed_ && auxPressed_ && (millis() - holdStartTime) >= TWO_BUTTONS_X_LONG_PUSH) {
+      switchModes(this);               // Pass "this" MultiProp instance to switchModes & switch the prop mode

```

Comment:
This comment is useless. The code already says the same thing.
A lot of comments in this code explains what the code does, but that's not the purpose of comments. The comments should focus on explaining *why* the code does what it does.
In most cases, what the code does is obvious from reading the code.